fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/ScalarOperatorsReuse.java

```diff
@@ -307,12 +315,115 @@ private boolean isNonDeterministicFuncOrLambdaArgumentExist(ScalarOperator scala
             }
 
             for (ScalarOperator child : scalarOperator.getChildren()) {
-                if (isNonDeterministicFuncOrLambdaArgumentExist(child, lambda)) {
+                if (isNonDeterministicFuncOrLambdaArgumentExist(child)) {
                     return true;
                 }
             }
             return false;
         }
 
     }
+
+
+    public static Projection getNewProjection(Projection projection, ColumnRefFactory columnRefFactory,
```

Comment:
What is the function of getNewProjection? Is it to calculate common expr? I think the name is a bit too vague, please add a comment and think about a more suitable name?

Reply:
renamed to `rewriteProjectionOrLambdaExpr`